webpack/assets/javascripts/react_app/redux/reducers/factCharts/index.js

```diff
@@ -0,0 +1,45 @@
+import Immutable from 'seamless-immutable';
+import {
+  FACT_CHART_DATA_REQUEST,
+  FACT_CHART_DATA_SUCCESS,
+  FACT_CHART_DATA_FAILURE,
+  CLOSE_FACT_CHART_MODAL,
+  OPEN_FACT_CHART_MODAL,
+} from '../../consts';
+
+const initialState = Immutable({
+  modalToDisplay: {},
+  chartData: [],
+  loaderStatus: '',
+  hostsCount: null,
+});
+
+let hostsCount = 0;
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FACT_CHART_DATA_REQUEST:
+      return state.set('loaderStatus', 'PENDING');
+    case FACT_CHART_DATA_SUCCESS:
+      action.payload.values.forEach((val) => { (hostsCount += val[1]); });
```

Comment:
this is probably a good candidate to introduce https://github.com/reactjs/reselect - @sharvit @amirfefer what do you guys think?

Reply:
@sharvit @amirfefer @waldenraines @tstrachota  ping?

Reply:
@lizagilman can you please create a follow up issue on redmine so we don't miss this?